tests/remove_point_evaluation/vector_tools_evaluate_at_points_01.cc

```diff
@@ -0,0 +1,272 @@
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// Interpolate solution between non-matching grids.
+
+#include <deal.II/base/quadrature_lib.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/fe_point_evaluation.h>
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/mapping_fe.h>
+#include <deal.II/fe/mapping_q_generic.h>
+
+#include <deal.II/grid/grid_generator.h>
+
+#include <deal.II/lac/la_parallel_vector.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/solver_control.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/numerics/vector_tools_evaluate.h>
+
+#include "../tests.h"
+
+using namespace dealii;
+
+namespace dealii
+{
+  namespace GridGenerator
+  {
+    template <int dim>
+    void
+    create_annulus(Triangulation<dim> &tria, const unsigned int n_refinements)
+    {
+      // according to the description in A FLEXIBLE, PARALLEL, ADAPTIVE
+      // GEOMETRIC MULTIGRID METHOD FOR FEM (Clevenger, Heister, Kanschat,
+      // Kronbichler): https://arxiv.org/pdf/1904.03317.pdf
+
+      GridGenerator::hyper_cube(tria, -1.0, +1.0);
+
+      if (n_refinements == 0)
+        return;
+
+      for (int i = 0; i < static_cast<int>(n_refinements) - 3; i++)
+        tria.refine_global();
+
+      if (n_refinements >= 1)
+        {
+          for (auto cell : tria.active_cell_iterators())
+            if (cell->is_locally_owned())
+              if (cell->center().norm() < 0.55)
+                cell->set_refine_flag();
+          tria.execute_coarsening_and_refinement();
+        }
+
+      if (n_refinements >= 2)
+        {
+          for (auto cell : tria.active_cell_iterators())
+            if (cell->is_locally_owned())
+              if (0.3 <= cell->center().norm() && cell->center().norm() <= 0.43)
+                cell->set_refine_flag();
+          tria.execute_coarsening_and_refinement();
+        }
+
+      if (n_refinements >= 3)
+        {
+          for (auto cell : tria.active_cell_iterators())
+            if (cell->is_locally_owned())
+              if (0.335 <= cell->center().norm() &&
+                  cell->center().norm() <= 0.39)
+                cell->set_refine_flag();
+          tria.execute_coarsening_and_refinement();
+        }
```

Comment:
I suggest to remove these two additional refinements - I think it's enough if you simply add some step to refine that creates sufficiently different parallel partitions in terms of the owned cells not overlapping. This keeps the test shorted (and the compactness is really the key of this class).